department-of-veterans-affairs/vets-website
# Add CustomEvent polyfill from Mozilla docs

## Description

This PR adds a polyfill for `CustomEvent` since it isn't supported natively in IE11. This API is necessary for the analytics coming from `component-library` components.

The polyfill was copy-pasted with light modifications from https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#polyfill
We are using `core-js` for several other polyfills, but that package [has chosen not to support `CustomEvent`](https://github.com/zloirock/core-js/issues/354).

Relevant [slack thread](https://dsva.slack.com/archives/CBU0KDSB1/p1620069518248000).


## Testing done

1. @U-DON narrowed the problem in `vets-website` down to a dependency upgrade for `component-library` from `2.3.1` to `2.4.0`. [changes](https://github.com/department-of-veterans-affairs/component-library/commit/6595ba4aa17a302f5d60dfda6a42c57d67cdea62#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519)
1. Confirmed that the problem was with the [`dispatchAnalyticsEvent` call](https://github.com/department-of-veterans-affairs/component-library/commit/6595ba4aa17a302f5d60dfda6a42c57d67cdea62#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519) by commenting it out and putting a simple `console.log('hi')` in the lifecycle method - the hca app loaded without the `Object does not support this action` error and we saw the expected log message.
1. Uncomment the event code and rebuild the dependency.
1. Add the polyfill in `vets-website`
1. Try to load an app and verify that it loads as expected
1. Put another `console.log` in [the analytics event handler](https://github.com/department-of-veterans-affairs/vets-website/blob/2d34a4ada4f19570d75caf863ed952de417b8267/src/platform/site-wide/component-library-analytics-setup.js#L26) to verify that the CustomEvent is being fired.


## Screenshots


## Acceptance criteria

## Definition of done
